pom.xml

```diff
@@ -1,101 +1,100 @@
-        <developer>
-            <id>timols</id>
-            <name>Tim Olshansky</name>
-            <email>tim.olshansky@gmail.com</email>
-        </developer>
-    </developers>
+  <scm>
+    <connection>scm:git:https://github.com/${gitHubRepo}.git</connection>
+    <developerConnection>scm:git:https://git@github.com/${gitHubRepo}.git</developerConnection>
+    <tag>${scmTag}</tag>
+    <url>https://github.com/${gitHubRepo}</url>
+  </scm>
 
-    <properties>
-        <!-- which version of Jenkins is this plugin built against? -->
-        <java.level>8</java.level>
-        <jenkins.version>2.121</jenkins.version>
-        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <java.compiler.version>1.8</java.compiler.version>
-      </properties>
-
-    <repositories>
-        <repository>
-            <id>repo.jenkins-ci.org</id>
-            <url>http://repo.jenkins-ci.org/public/</url>
-        </repository>
-    </repositories>
+  <properties>
+    <revision>2.1.0</revision>
+    <changelist>-SNAPSHOT</changelist>
+    <gitHubRepo>timols/jenkins-gitlab-merge-request-builder-plugin</gitHubRepo>
+    <!-- which version of Jenkins is this plugin built against? -->
+    <jenkins.version>2.452.3</jenkins.version>
+  </properties>
 
+  <dependencyManagement>
     <dependencies>
-        <!-- JMockit's dependency must be declared before that of JUnit. -->
-        <dependency>
-            <groupId>org.jmockit</groupId>
-            <artifactId>jmockit</artifactId>
-            <version>1.22</version>
-            <scope>test</scope>
-        </dependency>
-        <dependency>
-            <groupId>junit</groupId>
-            <artifactId>junit</artifactId>
-            <version>4.12</version>
-            <scope>test</scope>
-        </dependency>
-        <dependency>
-            <groupId>org.hamcrest</groupId>
-            <artifactId>hamcrest-all</artifactId>
-            <version>1.3</version>
-            <scope>test</scope>
-        </dependency>
-        <dependency>
-            <groupId>com.google.code.gson</groupId>
-            <artifactId>gson</artifactId>
-            <version>2.8.5</version>
-        </dependency>
-        <dependency>
-            <groupId>org.gitlab</groupId>
-            <artifactId>java-gitlab-api</artifactId>
-            <version>4.0.0</version>
-        </dependency>
+      <dependency>
+        <groupId>io.jenkins.tools.bom</groupId>
+        <artifactId>bom-2.452.x</artifactId>
+        <version>3814.v9563d972079a_</version>
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
     </dependencies>
+  </dependencyManagement>
+
+  <dependencies>
+    <dependency>
+      <groupId>com.google.code.gson</groupId>
+      <artifactId>gson</artifactId>
+      <version>2.8.5</version>
+    </dependency>
+    <dependency>
+      <groupId>org.gitlab</groupId>
+      <artifactId>java-gitlab-api</artifactId>
+      <version>4.0.0</version>
+    </dependency>
```

Comment:
Use https://plugins.jenkins.io/gitlab-api/dependencies/

Reply:
Thanks @basil . This change 
```
 <dependency>
      <groupId>io.jenkins.plugins</groupId>
      <artifactId>gitlab-api</artifactId>
    </dependency>
```

causes a bunch of issues. for example:

```
] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /Users/shlomodahan/Desktop/jenkins_plugins/jenkins-gitlab-merge-request-builder-plugin/src/main/java/org/jenkinsci/plugins/gitlab/Gitlab.java:[5,22] package org.gitlab.api does not exist
[ERROR] /Users/shlomodahan/Desktop/jenkins_plugins/jenkins-gitlab-merge-request-builder-plugin/src/main/java/org/jenkinsci/plugins/gitlab/Gitlab.java:[6,29] package org.gitlab.api.models does not exist
[ERROR] /Users/shlomodahan/Desktop/jenkins_plugins/jenkins-gitlab-merge-request-builder-plugin/src/main/java/org/jenkinsci/plugins/gitlab/Gitlab.java:[7,29] package org.gitlab.api.models does not exist
[ERROR] /Users/shlomodahan/Desktop/jenkins_plugins/jenkins-gitlab-merge-request-builder-plugin/src/main/java/org/jenkinsci/plugins/gitlab/Gitlab.java:[14,13] cannot find symbol
  symbol:   class GitlabAPI
  location: class org.jenkinsci.plugins.gitlab.Gitlab
[ERROR] /Users/shlomodahan/Desktop/jenkins_plugins/jenkins-gitlab-merge-request-builder-plugin/src/main/java/org/jenkinsci/plugins/gitlab/Gitlab.java:[25,12] cannot find symbol
  symbol:   class GitlabAPI
```
likely related to how we are importing the package. 

```
import org.gitlab.api.GitlabAPI;
import org.gitlab.api.models.GitlabCommitStatus;
import org.gitlab.api.models.GitlabProject;
```

What is the correct approach for handling this?

Reply:
See https://github.com/jenkinsci/gitlab-oauth-plugin/pull/66.